index.js

```diff
@@ -42,26 +42,24 @@ SuperCluster.prototype = {
 
         this.points = points;
 
-        // generate a cluster object for each point
+        // generate a cluster object for each point and index input points into a KD-tree
         var clusters = points.map(createPointCluster);
+        this.trees[this.options.maxZoom + 1] = kdbush(clusters, getX, getY, this.options.nodeSize, Float32Array);
+
         if (log) console.timeEnd(timerId);
 
         // cluster points on max zoom, then cluster the results on previous zoom, etc.;
         // results in a cluster hierarchy across zoom levels
         for (var z = this.options.maxZoom; z >= this.options.minZoom; z--) {
             var now = +Date.now();
 
-            // index input points into a KD-tree
-            this.trees[z + 1] = kdbush(clusters, getX, getY, this.options.nodeSize, Float32Array);
-
-            clusters = this._cluster(clusters, z); // create a new set of clusters for the zoom
+            // create a new set of clusters for the zoom and index input points into a KD-tree
```

Comment:
Since we index new clusters and not input points at this stage, we could say:

> create a new set of clusters for the zoom and index them with a KD-tree